app/scripts/views/search/location-search-directive.js

```diff
@@ -23,9 +24,15 @@
 
             ctl.clearError = clearError;
             ctl.onSearchButtonClicked = onSearchButtonClicked;
+            ctl.onTextKeyup = onTextKeyup;
         }
 
         function search(searchText, magicKey, options) {
+            if (searching) {
```

Comment:
I think it might be better to `_.throttle` with both leading and trailing edge calls.  Otherwise there might be some weird state where somebody hits enter, then sees results from a previous search.